ggml/ggml_convert.py

```diff
@@ -6,54 +6,330 @@
+        langs = card.field("langs").as_list(str)
+
+        default_lang = card.field("default_lang").as_(str)
+
+        return NllbLikeTokenizer(pathname, langs, default_lang)
+
+
+def convert_state_dict(
+    state_dict: Dict[str, Any], key_map: Optional[Mapping[str, str]] = None
+) -> Dict[str, Any]:
+
+    if key_map is None:
+        return state_dict
+    
+    state_dict = convert_model_state_dict(state_dict, key_map=key_map)
+
+    # We use the built-in version attribute of `torch.nn.Module`.
+    try:
+        del state_dict["encoder.version"]
+    except KeyError:
+        pass
+    try:
+        del state_dict["decoder.version"]
+    except KeyError:
+        pass
+
+    try:
+        del state_dict["encoder.embed_positions._float_tensor"]
+    except KeyError:
+        pass
+    try:
+        del state_dict["decoder.embed_positions._float_tensor"]
+    except KeyError:
+        pass
+
+    return state_dict
+
+
+def convert_unity_model(
+    model_name: str,
+    hparams: Optional[Dict[str, Any]] = None,
+):
+    from seamless_communication.models import unity
+    from seamless_communication.models.unity.builder import UnitYConfig, create_unity_model
+    from seamless_communication.models.unity.model import UnitYModel
+
+    load_unity_model_without_conversion = ModelLoader[UnitYModel, UnitYConfig](
+        asset_store,
+        download_manager,
+        unity.load_unity_config,
+        create_unity_model,
+        None,
+        restrict_checkpoints=False,
+    )
+
+    model_config = unity.load_unity_config(model_name)
+    hparams = flatten_config(
+        dataclasses.asdict(model_config), separator="__", overrides=hparams
+    )
+    hparams["multilingual"] = True
+    log.info(hparams)
+    # Need the diverge here because current default in SC is to convert from fairseq1 ckpt format
+    if model_name in UNITY_SMALLER_MODELS:
+        model = load_unity_model_without_conversion(model_name)
+        tokenizer = NllbLikeTokenizerLoader(asset_store, download_manager)(model_name)
+    else:
+        model = unity.load_unity_model(model_name)
+        tokenizer = unity.load_unity_text_tokenizer(model_name)
+
+    vocab = read_vocab(tokenizer)
+
+    return model, hparams, vocab
+
+
+def convert_nllb_model(
+    model_name: str,
+    hparams: Optional[Dict[str, Any]] = None,
+):
+    from fairseq2.models.nllb.loader import load_nllb_tokenizer, load_nllb_model, load_nllb_config
+
+    model_config = load_nllb_config(model_name)
+    hparams = flatten_config(
+        dataclasses.asdict(model_config), separator="__", overrides=hparams,
+    )
+    hparams["multilingual"] = True
+
+    model = load_nllb_model(model_name)
+    tokenizer = load_nllb_tokenizer(model_name)
+    vocab = read_vocab(tokenizer)
+
+    return model, hparams, vocab
+
+
+def convert_bitext_model(
+    model_name: str,
+    hparams: Optional[Dict[str, Any]] = None,
+):
+    from mt import load_mt_model, load_vocab  #, test_mt
```

Comment:
I think it is fine to upstream mt.py here . Its general purpose is to support backward compatibilty for fairseq-based text translation models that use standard transformer. 

Either that, or we skip this function (and trim the permissible module type above)